packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1100,6 +1114,11 @@ class SelectionOverlay {
   final ValueChanged<DragEndDetails>? onStartHandleDragEnd;
 
   void _handleStartHandleDragEnd(DragEndDetails details) {
+    // Calling OverlayEntry.remove may not happen until the following frame, so
+    // it's possible for the handles to receive a gesture after calling remove.
+    if (_handles == null) {
```

Comment:
Would this result in a unbalanced call? You're handling a handle drag event and expects an end event at some point,  but you never receives an end event because the handle being dragged becomes invisible?

Reply:
That's true, but it's the same as the current behavior.  _handleStart/EndHandleDragEnd is never called when the bug is reproduced.

I just tried it and it looks like the same thing will happen if you suddenly remove a GestureDetector from the Widget tree:

<details>

<summary>Example of removing a GestureDetector while it's detecting a drag. Drag end is never called.</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            DisappearingWidget(),
          ],
        ),
      ),
    );
  }
}

class DisappearingWidget extends StatefulWidget {
  const DisappearingWidget({super.key});

  @override
  State<DisappearingWidget> createState() => _DisappearingWidgetState();
}

class _DisappearingWidgetState extends State<DisappearingWidget> {
  bool detecting = true;

  @override
  Widget build(BuildContext context) {
    const Text child = Text('Drag me');
    if (!detecting) {
      return child;
    }
    return GestureDetector(
      onHorizontalDragStart: (DragStartDetails details) {
        print('justin drag');
        Future<void>.delayed(const Duration(seconds: 2)).then((_) {
          print('justin turning off detection');
          setState(() {
            detecting = false;
          });
        });
      },
      onHorizontalDragUpdate: (DragUpdateDetails details) {
        print('justin drag update');
      },
      onHorizontalDragEnd: (DragEndDetails details) {
        print('justin drag end');
      },
      child: child,
    );
  }
}
```

</details>

I did just change things around so that _isDraggingStart/EndHandle are always updated though, even if this `if` is tripped.

Reply:
You'll get a drag cancel I think?

Reply:
I just tried it again with a cancel handler and it was never called.  I've edited my previous comment with the example code to include the cancel handler.

I wonder if that's a bug and we should be calling cancel in the `dispose` method...

For completeness I also tried reproducing the bug on master with a cancel handler and the cancel handler also was not called, as expected.

Reply:
I'm merging this but I'm happy to revisit anything if you want.